### PR TITLE
Feature/read excel template

### PR DESCRIPTION
**Pull request by AsafSH6**
Read excel as template with styles and apply the data of given DataFrame to it.

### CODE REVIEW

File: StyleFrame/style_frame.py [hunk 1]
```diff
@@ -125,25 +126,26 @@ def __getattr__(self, attr):
...
-        elif isinstance(column_to_convert, int) and column_to_convert >= 1:
-            column_as_letter = cell.get_column_letter(startcol + column_to_convert)
+        elif isinstance(col, int) and col >= 1:
+            column_as_letter = cell.get_column_letter(startcol + col)
 
         # assuming we got column letter
-        elif isinstance(column_to_convert, (str_type, unicode_type)) < get_column_letter(sheet.max_column):
-            column_as_letter = column_to_convert
-
-        if column_as_letter is None or cell.column_index_from_string(column_as_letter) > sheet.max_column:
```
[the remark this answers is not in the document]
Comment (DeepSpace2) [reply StyleFrame/style_frame.py:145]: I think this validation should stay, and instead, we should update the height/width dicts in run time (the styleframe only changes based on the `use_df_boundaries` )

File: StyleFrame/style_frame.py [hunk 2]
```diff
@@ -399,8 +447,6 @@ def get_range_of_cells(row_index=None, columns=None):
         for row in self._rows_height:
             if within_sheet_boundaries(row=(row + startrow)):
                 sheet.row_dimensions[startrow + row].height = self._rows_height[row]
-            else:
```
[the remark this answers is not in the document]
Comment (DeepSpace2) [reply StyleFrame/style_frame.py:402]: I think this validation should stay, and instead, we should update the height/width dicts in run time (the styleframe only changes based on the `use_df_boundaries` )

File: styleframe/style_frame.py [hunk 3]
```diff
@@ -249,7 +252,10 @@ def read_excel_as_template(cls, path, df, use_df_boundaries=False, **kwargs):
 
         if use_df_boundaries:
             sf.data_df = sf.data_df.iloc[:num_of_rows, :num_of_cols]
-
+            sf._rows_height = {k: v for i, (k, v) in enumerate(sf._rows_height.items())
```
Comment (AsafSH6) [inline styleframe/style_frame.py:255]: 1. sf._row_height is now OrderedDict not regular dict (you use dict comprehension not OrderedDict)
2. `sf._row_height = OrderDict(sf._rows_height.items()[:num_of_rows])`

Comment (AsafSH6) [reply styleframe/style_frame.py:255]: We should even consider to implement setter to `_row_height` and `_column_width` that validate this or convert dict to `OrderedDict`

File: styleframe/styler.py [hunk 4]
```diff
@@ -47,6 +46,12 @@ def get_color_from_string(color_str, default_color=None):
         self.comment_text = comment_text
         self.text_rotation = text_rotation
 
+        if self.border_type == utils.borders.default_grid:
```
Comment (AsafSH6) [inline styleframe/styler.py:49]: It should be as part of `border_type` setter instead

**Pull request merged**